src/shared_utilities/models.jl

```diff
@@ -317,7 +317,7 @@ end
 function initialize_vars(keys, types, domain_names, state, model_name)
     FT = eltype(state)
     if length(keys) == 0
-        return (; model_name => FT[])
+        return (; model_name => nothing)
```

Comment:
With these changes, `nothing` is only filtered out for canopy models, but any model could potentially have `nothing` in `Y` or `p` since they all use this function. Is this okay, or should we call `filter_nt` in every case? It might be nice to avoid as long as we can, since it will add more allocations

Reply:
darn! and I guess we cant easily change `filter_nt` so that if there are no `nothing`s, it doesnt re-allocate? I kind of think we want to call it in every case to cover our bases...

Reply:
Not sure if this is too much, but we could write a function which recursively gets the values of all keys in the NT. If none of them are `nothing` we can do nothing, but if some are, we `filter_nt`?